File: system-test/typescript/src/testkit/driver/adapter/api/client/MyShopApiClient.ts
import { HealthController } from './controllers/HealthController.js';
import { OrderController } from './controllers/OrderController.js';
import { CouponController } from './controllers/CouponController.js';

export class MyShopApiClient {
  private readonly healthController: HealthController;
  private readonly orderController: OrderController;
  private readonly couponController: CouponController;

  constructor(baseUrl: string) {
    this.healthController = new HealthController(baseUrl);
    this.orderController = new OrderController(baseUrl);
    this.couponController = new CouponController(baseUrl);
  }

  health(): HealthController {
    return this.healthController;
  }

  orders(): OrderController {
    return this.orderController;
  }

  coupons(): CouponController {
    return this.couponController;
  }
}

File: system-test/dotnet/Driver.Adapter/Api/Client/MyShopApiClient.cs
using Driver.Adapter.Shared.Client.Http;
using Driver.Adapter.Api.Client.Dtos.Errors;
using Driver.Adapter.Api.Client.Controllers;

namespace Driver.Adapter.Api.Client;

public class MyShopApiClient : IDisposable
{
    private readonly JsonHttpClient<ProblemDetailResponse> _httpClient;
    private readonly HealthController _healthController;
    private readonly OrderController _orderController;
    private readonly CouponController _couponController;
    private bool _disposed;

    public MyShopApiClient(string baseUrl)
    {
        _httpClient = new JsonHttpClient<ProblemDetailResponse>(baseUrl);
        _healthController = new HealthController(_httpClient);
        _orderController = new OrderController(_httpClient);
        _couponController = new CouponController(_httpClient);
    }

    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }

    protected virtual void Dispose(bool disposing)
    {
        if (_disposed) return;
        if (disposing)
            _httpClient?.Dispose();
        _disposed = true;
    }

    public HealthController Health() => _healthController;

    public OrderController Orders() => _orderController;

    public CouponController Coupons() => _couponController;
}
